src/main/java/com/bigstep/datalake/DLFileSystem.java

```diff
@@ -1559,6 +1607,92 @@ Boolean decodeResponse(Map<?, ?> json) throws IOException {
         }
     }
 
+    class EncryptedFsPathOutputStreamRunner extends AbstractFsPathRunner<FSDataOutputStream> {
+        private final int bufferSize;
+
+        EncryptedFsPathOutputStreamRunner(Op op, Path fspath, int bufferSize,
+                                 Param<?, ?>... parameters) {
+            super(op, fspath, parameters);
+            this.bufferSize = bufferSize;
+        }
+
+        @Override
+        FSDataOutputStream getResponse(final HttpURLConnection conn)
+                throws IOException {
+            int nHeaderSize = DLEncryptionUtils.getHeaderDetailLength();
+            byte[] b = new byte[nHeaderSize];
+            byte[] key = DLEncryptionUtils.getSecretKey();
+            int nBlockSize = key.length;
+            byte[] initVector = new byte[nBlockSize];
+
+            if (this.op == PostOpParam.Op.APPEND)
+            {
+                // get IV
+                FsPathConnectionRunner runnerHeader =
+                        new FsPathConnectionRunner(GetOpParam.Op.OPEN, this.getFSPath(), new BufferSizeParam(bufferSize), new LengthParam((long) nHeaderSize + nBlockSize));
+                OffsetUrlInputStream inputStreamHeader = new OffsetUrlInputStream(new UnresolvedUrlOpener(runnerHeader), new OffsetUrlOpener(null));
+
+                inputStreamHeader.readFully(0, b, 0, nHeaderSize);
+                inputStreamHeader.readFully(nHeaderSize, initVector, 0, nBlockSize);
+
+                //get fileSize
+                long fileSize;
+                HdfsFileStatus fileStatus = getHdfsFileStatus(this.getFSPath());
+                fileSize = fileStatus.getLen();
+
+                BufferedOutputStream outputStream = new BufferedOutputStream(conn.getOutputStream(), bufferSize);
+
+                Properties props = new Properties();
+                props.put("commons.crypto.stream.buffer.size", bufferSize);
+
+                CtrCryptoOutputStream cryptoOutputStream = new CtrCryptoOutputStream(props, outputStream, key, initVector, fileSize - nHeaderSize - nBlockSize);
+                return new FSDataOutputStream(cryptoOutputStream, statistics) {
+                    @Override
+                    public void close() throws IOException {
+                        try {
+                            super.close();
+                        } finally {
+                            try {
+                                validateResponse(op, conn, true);
+                            } finally {
+                                conn.disconnect();
+                            }
+                        }
+                    }
+                };
+            }
+            else
+            {
+                try {
+                    initVector = DLEncryptionUtils.generateRandomIV();
+                }
+                catch (Exception ex) {
+                    // TODO
```

Comment:
You should log the error and  return from the function. All exceptions should be treated.